From 6bf660a04acccd705a8ddb0732db3e6b0c72aecf Mon Sep 17 00:00:00 2001
From: Li Wang <li.wang@windriver.com>
Date: Mon, 8 Jul 2013 11:13:12 +0800
Subject: [PATCH 5/7] x86: compile warning: different variable type comparison

warning: comparison between signed and unsigned integer expressions [-Wsign-compare]

Signed-off-by: Li Wang <li.wang@windriver.com>
---
 arch/x86/boot/compressed/mkpiggy.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/boot/compressed/mkpiggy.c b/arch/x86/boot/compressed/mkpiggy.c
index 958a641..f24c757 100644
--- a/arch/x86/boot/compressed/mkpiggy.c
+++ b/arch/x86/boot/compressed/mkpiggy.c
@@ -70,7 +70,7 @@ int main(int argc, char *argv[])
 	 * sizes, compute the necessary decompression offset...
 	 */
 
-	offs = (olen > ilen) ? olen - ilen : 0;
+	offs = ((long)olen > ilen) ? olen - ilen : 0;
 	offs += olen >> 12;	/* Add 8 bytes for each 32K block */
 	offs += 64*1024 + 128;	/* Add 64K + 128 bytes slack */
 	offs = (offs+4095) & ~4095; /* Round to a 4K boundary */
-- 
1.7.10.4

